src/ray/raylet/worker_pool.h

```diff
@@ -39,6 +40,56 @@ namespace raylet {
 using WorkerCommandMap =
     std::unordered_map<Language, std::vector<std::string>, std::hash<int>>;
 
+/// \struct WorkerCacheKey
+///
+/// Struct used as a cache key for workers. The dynamic_options and
+/// override_environment_variables will be unified into runtime_env in the future.
+struct WorkerCacheKey {
+  TaskID task_id;
+  std::vector<std::string> dynamic_options;
+  std::unordered_map<std::string, std::string> override_environment_variables;
+  std::string serialized_runtime_env;
+  mutable std::size_t hash_ = 0;
+
+  bool operator==(const WorkerCacheKey &k) const {
```

Comment:
Should we just compare the hashes?

Reply:
I don't think so, implementing operator== is a requirement for hashable types specifically to handle the case when the hashes are equal, so that would defeat the purpose. (according to https://stackoverflow.com/a/17017281/14651138)

Reply:
Cool, makes sense!